docs/design/adrs/2024-05-02-enable-project-versioning.md

```diff
@@ -0,0 +1,56 @@
+# Use Github Releases to manage project versioning
+
+* **Status:** proposed
+* **Proposer:** @frtibble
+* **Date:** 2024-05-02
+* **Technical Story:** [https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/651](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/651)
+
+## Context and Problem Statement
```

Comment:
nitpick: in this section we discuss the benefits of "releases" but we dont mention what is our problem that we'll solve with releases.

Reply:
Perhaps I can help with this. The problem we're facing currently is that we don't have a way to communicate to the users of this repo which changes are "breaking" - for example, which changes would recreate the search index from scratch, or which changes would make the current configuration incompatible with the latest deployment. This is causing customers to pull the latest changes only to see undesired behavior (maybe they aren't ready to recreate their index yet).

Versioning/formal releases will help us communicate with our customers what changes went in, and is it a bugfix/feature and/or a breaking change that they need to be cautious about.